### PR TITLE
Added Onboarding Unit Tests, Added Error Handing to the Onboarding page, Fixed an issue with the AuthenicationController, and Added a feature to the SignUp unit test which deletes the created user after each test

### PR DESCRIPTION
**Pull request by rushanniazi**
- Added Onboarded Unit tests for multiple edge cases
- Added error handling to onboarding page to not allow user to continue if required fields are not filled in
- Also user must input a valid email address and an age > 13 to sign up for an account
- Added functionality to SignUp unit test to remove the created test user from Users table when the test is complete

### CODE REVIEW

File: test/unitTests.js [hunk 1]
```diff
@@ -1,32 +1,52 @@
 const RecommendController = require('../travvy-server/src/controllers/RecommendController.js')
 const AuthenticationController = require('../travvy-server/src/controllers/AuthenticationController.js')
```
Comment (rushanniazi) [inline test/unitTests.js:3]: I just moved around some of the lines of code to make the unitTests.js file more organized for each type of test

File: test/unitTests.js [hunk 2]
```diff
@@ -38,7 +58,7 @@ describe('Incorrect email and correct password log in test', function() {
           assert.equal(response,null);
            })
        });
-
+describe('Incorrect email and incorrect password log in test', function() {
```
Comment (lkdfount) [inline test/unitTests.js:61]: This is a duplicate. It must have got messed up while we were fixing conflicts for a different merge. I will remove it in my next PR because I already need to make a few changes to this.

File: test/unitTests.js [hunk 3]
```diff
@@ -65,31 +85,49 @@ describe('Correct email and correct password log in test', function() {
...
         })
+    });
+
+
+// Unit Tests for Home Page
+const assert = require('assert');
+const city = "Toronto"
+const groupSize = 20
+const startDate = new Date("2020-11-05")
+const endDate = new Date("2021-01-05")
```
Comment (lkdfount) [inline test/unitTests.js:96]: Thanks for organizing this. It makes it easier to read.

File: test/unitTests.js [hunk 4]
```diff
@@ -65,31 +85,49 @@ describe('Correct email and correct password log in test', function() {
...
-    })
-});
     });
 
+describe('Onboarding Test for Incorrect email and age, however correct gender, instagram username, and preferred activity type', function() {
+ it('Onboarding test should return false since data for email and age are incorrect', async function() {
+       const response = await AuthenticationController.onboardingTest(incorrectOnboardingEmail,unacceptableAge,acceptableGender,acceptableInstagramUsername,acceptableActivity)
+       assert.equal(response, null);
+        })
+    });
```
Comment (lkdfount) [inline test/unitTests.js:130]: Should we test for other unacceptable values as well? Why are we only testing for unacceptable email and age?
Also, could we test with just unacceptable age and a correct email and one with acceptable age but unacceptable email?

File: travvy/src/services/AuthenticationService.js [hunk 5]
```diff
@@ -43,13 +43,26 @@ export default {
         return isValidLogin
     },
     async onboarding(email,age,gender,instagram_username,preferred_activity_type){
-        return axios.post('/onboarding', {params:email,age,gender,instagram_username,preferred_activity_type})  
+       
+       var checkEmail = false
```
Comment (lkdfount) [inline travvy/src/services/AuthenticationService.js:47]: Good idea to check the email is valid!

File: test/unitTests.js (hunk 4 above)
Comment (rushanniazi) [reply test/unitTests.js:130]: Great idea, I am hopefully planning on extending these unit tests to consider all edge cases such are you mentioned

File: travvy/src/components/Onboarding.vue [hunk 6]
```diff
@@ -4,19 +4,27 @@
...
+  <form>
     <h3>Please fill in the following information in order to get a custom travel experience.</h3>
     <br>
+    <p v-if="errors.length">
+    <b class="red">Please correct the following error(s):</b>
+    <!-- this will bind all the errors together after the form is completed and return them to the user-->
+      <u1 class="red">
+        <li v-bind="error in errors"> {{ errors }}</li>
+      </u1>
+    </p>
```
Comment (syasrebii) [inline travvy/src/components/Onboarding.vue:17]: Perfect, thanks so much for adding this error handling in, this will definitely help users identify whether their input follows the correct format going forward. Out of curiosity, this would just return a list of errors of places the user inputted incorrect format/info at the top of the page, correct?

File: travvy/src/components/Onboarding.vue [hunk 7]
```diff
@@ -4,19 +4,27 @@
...
     <br>
-
-    <label>Please input your email here: </label>
-    <input type="text" class="info" v-model="email">
+    <label><strong>Please verify your email here (required): </strong></label>
+    <input type="text" class="info" v-model="email" placeholder="Input Email here" required>
```
Comment (syasrebii) [inline travvy/src/components/Onboarding.vue:20]: Good thinking with adding the "strong" HTML element for required information, great indicator so it improves our user experience!

File: travvy/src/components/Onboarding.vue [hunk 8]
```diff
@@ -130,6 +150,53 @@ export default {
             } catch (error) {
                console.log(error)
             }
+      },
+
+      checkForm: function(x){
+              // an array with the current errors and what will be returned to the user
+              this.errors = [];
```
Comment (syasrebii) [inline travvy/src/components/Onboarding.vue:157]: We finally have error handling for all input fields!! Out of curiosity, do we want to explicitly indicate whether an input will be optional/"if applicable" (i.e. dream vacation spot or Instagram handles)? Maybe that's something we could quickly add on the frontend.

File: test/unitTests.js [hunk 9]
```diff
@@ -1,32 +1,52 @@
...
+        assert.equal(response.length,user.length);
 
-describe('Recommend Attractions Test', function() {
- it('Recommend Attractions should return 4', async function() {
-       const response = await RecommendController.recommend(city,groupSize,startDate,endDate)
-       assert.equal(response.length,4);
+        const deleteUser = await Users.destroy({
+            where:{
+                Email:email,
+            }
```
Comment (emilynairnn) [inline test/unitTests.js:36]: This is a great addition, now we will not have to manually delete our test cases from the database. Thanks!

File: travvy/src/components/Onboarding.vue (hunk 6 above)
Comment (rushanniazi) [reply travvy/src/components/Onboarding.vue:17]: Correct, I implemented similar functionality as the SignUp page's error handling in order to keep consistent UI and allow the user to enjoy a positive user experience

File: travvy/src/components/Onboarding.vue (hunk 8 above)
Comment (rushanniazi) [reply travvy/src/components/Onboarding.vue:157]: Yeah that is definitely an option, for now I just added 'required' beside the fields which we consider to be mandatory
